### ISSUE TITLE
Provider not ready error occur in Mac Safari browser

### ISSUE THREAD

**Issue by parthipanr** [abacritt/angularx-social-login#371]
I have integrated this package in my application for google and Facebook login. It was working properly in all browser except Safari browser in mac environment. I have checked your demo application in mac safari browser and faced the same error. Please refer the below screenshot. 
![image](https://user-images.githubusercontent.com/15604974/108979965-3414cf00-76b1-11eb-978e-f02ec74c3c28.png)

**Comment by titusfx** [comment]
I have the same error, in Chrome. I just copy and paste the example. Sometimes works and sometimes does not. I believe that is, the way, the library use the Providers, the gapi didn't resolve completely or throw an error and the library assume everything is ok. Or angular tries to Init the providers, but the gapi didn't resolve

**Comment by selvin007** [comment]
> I have integrated this package in my application for google and Facebook login. It was working properly in all browser except Safari and chrome browser in mac environment. I have checked your demo application in mac safari browser and faced the same error. Please refer the below screenshot.
> ![image](https://user-images.githubusercontent.com/15604974/108979965-3414cf00-76b1-11eb-978e-f02ec74c3c28.png)

Check if third party cookies are disabled in settings
google link doesn't work with third party cookies disabled
src: https://developers.google.com/identity/sign-in/web/troubleshooting

**Comment by titusfx** [comment]
I assume this is error is due to injection. The value Is not ready at the moment is used. Is a library problem. You could track it to the source code.

**Comment by jaibatrik** [comment]
No, the library provides an `Observable` to broadcast when it's ready. Wait for the callback before enabling the sign in buttons.
